src/utils/storageService.ts

```diff
@@ -1,58 +1,52 @@
 import { openDb } from 'idb'
-import { dispatch } from './store'
-
 import { Question } from 'common/models'
 
-const dbPromise = openDb('form-db', 1, upgradeDB => {
-    upgradeDB.createObjectStore('formStore', {
-        keyPath: 'id',
-        autoIncrement: true,
+const StorageServiceFactory = (accessDB: any) => {
+    const dbPromise = accessDB('form-db', 1, (upgradeDB: any) => {
+        upgradeDB.createObjectStore('formStore', {
+            keyPath: 'id',
+            autoIncrement: true,
+        })
     })
-})
 
-export const storageService = {
-    async getAllQuestions() {
+    const getAllQuestions = async () => {
         const db = await dbPromise
         return db
             .transaction('formStore')
             .objectStore('formStore')
             .getAll()
-            .then(res => {
-                dispatch.form.initQuestionList(res)
-            })
-    },
+    }
 
-    async updateQuestion(updatedQuestion: Question) {
+    const updateQuestion = async (updatedQuestion: Question) => {
         const db = await dbPromise
         const tx = db.transaction('formStore', 'readwrite')
-        tx.objectStore('formStore')
-            .put(updatedQuestion)
-            .then(res => {
-                updatedQuestion.id = parseInt(res.toString(), 10)
-                dispatch.form.updateQuestion(updatedQuestion)
-            })
-        return tx.complete
-    },
+        return tx.objectStore('formStore').put(updatedQuestion)
```

Comment:
I see a hardcoded `'formStore'` multiple times - good thing is to store such values in the variables